### ISSUE TITLE
Final 3x3 conv missing for P3, P4, P5?

### ISSUE THREAD

**Issue by mhwilder** [delftrobotics/keras-retinanet#35]
I was looking through the model specification code and also looking back at the RetinaNet paper and the FPN paper and perhaps found a discrepancy. In the FPN [paper](https://arxiv.org/pdf/1612.03144.pdf) (first block of text on page 4), the authors mention using a 3x3 conv layer to reduce aliasing effects. It appears that this is not in the current implementation here and I'm wondering if there's a reason or if it was just overlooked. I realize that this would be somewhat redundant with first layers of the classification and regression submodels, but since it wouldn't be shared across levels of the pyramid, it could have utility.

I don't have a great sense of how important this layer is, but I thought it was at least worth bringing up. The RetinaNet paper is a bit lacking in description so it's not even totally clear that they reused this aspect of FPN.

**Comment by mhwilder** [comment]
Yeah, I'd agree that the wording in their paper suggests this should only be applied to P3 and P4.

**Comment by jwnsu** [comment]
Hope this helps close accuracy gap with the paper :-).

**Comment by mhwilder** [comment]
Yeah, I was thinking it could potentially help with that. Seems that training on batch size of 1 could also be affecting final accuracy. Any progress on removing that restriction?

**Comment by hgaiser** [comment]
Same here :) we'll see.

**Comment by hgaiser** [comment]
> Yeah, I was thinking it could potentially help with that. Seems that training on batch size of 1 could also be affecting final accuracy. Any progress on removing that restriction?

I suspect it will mainly help with training speed (which would dramatically increase development time). I'm not currently looking into the batch size issue, @florisdesmedt was looking into that. Feel free to join the Keras Slack channel (#keras-retinanet) for discussions on this topic.

**Comment by hgaiser** [comment]
The changes are on [this](https://github.com/delftrobotics/keras-retinanet/tree/add-missing-conv-fpn) branch by the way. I'll hold off on merging it to master until the model is done training. That way I can simultaneously update the model and the architecture. For now I'll close this issue.
